azkaban-common/src/main/java/azkaban/imagemgmt/daos/ImageTypeDaoImpl.java

```diff
@@ -122,6 +122,32 @@ public int createImageType(final ImageType imageType) {
     return imageTypeId;
   }
 
+  @Override
+  public ImageType getImageTypeWithOwnershipsById(final String id) {
+    final FetchImageTypeHandler fetchImageTypeHandler = new FetchImageTypeHandler();
+    List<ImageType> imageTypes = new ArrayList<>();
+    try {
+      imageTypes = this.databaseOperator.query(FetchImageTypeHandler.FETCH_IMAGE_TYPE_BY_ID,
+          fetchImageTypeHandler, id);
+      if (imageTypes == null || imageTypes.isEmpty()) {
+        throw new ImageMgmtDaoException(ErrorCode.NOT_FOUND,
+            "Failed to find image type for " + id);
+      }
+      if (imageTypes != null && imageTypes.size() > 1) {
+        throw new ImageMgmtDaoException(ErrorCode.UNPROCESSABLE_ENTITY, "The request for image "
+            + "type with id " + id + " unexpectedly returned more than one result!");
+      }
+    } catch (final SQLException ex) {
+      log.error(FetchImageTypeHandler.FETCH_IMAGE_TYPE_BY_ID + " failed.", ex);
+      throw new ImageMgmtDaoException(ErrorCode.NOT_FOUND,
+          "Unable to get image type for : " + id);
+    }
```

Comment:
If a SQLException occurred then I think we should return an Internal Server (500) error response to the user.

Reply:
I agree. Have changed them all for all calls in the class.